service/src/main/java/bio/terra/workspace/service/resource/controlled/ControlledResourceMetadataManager.java

```diff
@@ -86,17 +98,66 @@ public ControlledResource validateControlledResourceAndAction(
     WsmResource resource = resourceDao.getResource(workspaceUuid, resourceId);
-
     ControlledResource controlledResource = resource.castToControlledResource();
-    SamRethrow.onInterrupted(
-        () ->
-            samService.checkAuthz(
-                userRequest,
-                controlledResource.getCategory().getSamResourceName(),
-                resourceId.toString(),
-                action),
-        "checkAuthz");
-    return controlledResource;
+    String samName = controlledResource.getCategory().getSamResourceName();
+
+    // Every case exits the validation
+    switch (controlledResource.getState()) {
+      case READY:
+        // Primary success case: check auth and return the resource
+        SamRethrow.onInterrupted(
+            () -> samService.checkAuthz(userRequest, samName, resourceId.toString(), action),
+            "checkAuthz");
+        return controlledResource;
+
+      case CREATING, DELETING, UPDATING:
+        // Check forbidden before throwing the busy resource error
+        SamRethrow.onInterrupted(
+            () -> samService.checkAuthz(userRequest, samName, resourceId.toString(), action),
+            "checkAuthz");
+        throw new ResourceIsBusyException(
+            "Another operation is running on the resource; wait and try again");
+
+      case BROKEN_DELETE:
+        // If the resource is in the BROKEN_DELETING state, then it is in an unknown broken state
+        // and cannot be deleted by a user. It has to be cleaned up manually by Terra personnel.
+        throw new ResourceIsCorruptException(
+            "Resource is in an internally corrupted state. Contact support.");
+
+      case BROKEN:
+        // If the resource is in the BROKEN state, then there may be no Sam resource. We do our
+        // best with what we know. We handle two cases:
+        //  1. If the resource is an application resource, and the application is making the
+        // request,
+        //     we go ahead.
+        //  2. If the resource is a user resource, we test that the user is a WRITER on the
+        // workspace.
+        if (controlledResource.getManagedBy() == ManagedByType.MANAGED_BY_APPLICATION) {
+          WsmApplication application =
+              applicationDao.getApplication(controlledResource.getApplicationId());
+          String callerEmail = samService.getUserEmailFromSamAndRethrowOnInterrupt(userRequest);
+          if (!StringUtils.equals(application.getServiceAccount(), callerEmail)) {
+            throw new ForbiddenException(
+                "Only the associated application "
+                    + application.getDisplayName()
+                    + "can delete a broken application resource");
+          }
+        } else {
+          // broken user resource
+          SamRethrow.onInterrupted(
+              () ->
+                  samService.checkAuthz(
+                      userRequest,
+                      SamConstants.SamResource.WORKSPACE,
```

Comment:
why is the type WORKSPACE?

Reply:
I updated the comment. It is because the Sam resource creation might have failed or we will have executed the "UNDO" on the create; since the Sam resource doesn't exist, we cannot test authz against it. So we test against the workspace.